ntbs-service/Services/AuditService.cs

```diff
@@ -100,5 +103,20 @@ public async Task<IList<AuditLog>> GetWriteAuditsForNotification(int notificatio
                 .Where(log => log.RootId == notificationId.ToString())
                 .ToListAsync();
         }
+
+        public async Task AuditSearch(IQueryCollection queryParameters, string userName)
+        {
+            var parametersWithValuesDictionary = queryParameters
+                .Where(kvp => !string.IsNullOrEmpty(kvp.Value))
+                .ToDictionary(kvp => kvp.Key, kvp => kvp.Value.First());

```

Comment:
This code only works because we know that each of the query parameters has a single value. I think we should add a comment stating this assumption here.

Reply:
Have added a comment